### ISSUE TITLE
[NEWS] Przypomnienie: na początku 2024 roku wyłączymy zasoby /sale/offers i /sale/offers/{offerId} / Reminder: at the beginning of 2024, we will disable the /sale/offers resources for creating and editing offers

### ISSUE THREAD

**Issue by AureliuszBrussy** [allegro/allegro-api#6481]
Przypominamy, że [zgodnie z zapowiedzią](https://developer.allegro.pl/news/na-poczatku-2024-roku-wylaczymy-zasoby-sale-offers-sluzace-do-tworzenia-i-edycji-ofert-BvqK3XOaEcW) planujemy całkowicie wyłączyć zasoby do tworzenia oraz edycji ofert ze ścieżek **/sale/offers** oraz **/sale/offers/{offerId}**, dlatego:
- **1 stycznia 2023 roku** oznaczymy je w dokumentacji jako **Deprecated**,
- **1 stycznia 2024 roku** całkowicie je **usuniemy**.

W styczniu tego roku udostępniliśmy zasoby **/sale/product-offers** w wersji public, dzięki którym:
- [za pomocą jednego żądania wystawisz aktywną ofertę powiązaną z produktem](https://developer.allegro.pl/tutorials/jak-jednym-requestem-wystawic-oferte-powiazana-z-produktem-D7Kj9gw4xFA),
- [edytujesz dowolne pole oferty](https://developer.allegro.pl/tutorials/jak-zarzadzac-ofertami-7GzB2L37ase#edycja-pojedynczej-oferty),
- [utworzysz zestaw produktowy](https://developer.allegro.pl/tutorials/jak-jednym-requestem-wystawic-oferte-powiazana-z-produktem-D7Kj9gw4xFA#jak-utworzyc-zestaw-produktowy).

Zasoby **/sale/product-offers** będą jedynymi, które domyślnie umożliwią wystawienie oraz edycję oferty. Zadbaj o to, żeby przed całkowitym wyłączeniem zasobów **/sale/offers** Twoje oprogramowanie korzystało z odpowiedników tych zasobów na ścieżce **/sale/product-offers**: 

| **Endpoint, który usuniemy** | **Odpowiednik endpointu** | 
| :--------: | :--------: | 
| [POST /sale/offers](https://developer.allegro.pl/documentation#operation/createOfferUsingPOST) | [POST /sale/product-offers](https://developer.allegro.pl/documentation#operation/createProductOffers) | 
| [PUT /sale/offers/{offerId}](https://developer.allegro.pl/documentation#operation/updateOfferUsingPUT) | [PATCH /sale/product-offers/{offerId}](https://developer.allegro.pl/documentation#operation/editProductOffers) | 
| [GET /sale/offers/{offerId}](https://developer.allegro.pl/documentation#operation/getOfferUsingGET) | [GET /sale/product-offers/{offerId}](https://developer.allegro.pl/documentation#operation/getProductOffer) |

**Ważne!** **Nie wyłączymy** zasobu do odczytywania listy oferty ([GET /sale/offers](https://developer.allegro.pl/documentation#operation/searchOffersUsingGET)) - wciąż będzie on dostępny. 

Cały czas pracujemy nad rozwijaniem zasobów **/sale/product-offers**. Weryfikujemy oraz uzupełniamy brakujące funkcje względem **/sale/offers**, bądź wprowadzamy nowe zasoby, które zapewnią ciągłość dostępu do funkcjonalności. 

**Czy zasoby /sale/offers umożliwią wyświetlanie ofert i zarządzanie ofertami w zagranicznych domenach?**

Możliwość [wyświetlania ofert w zagranicznych domenach Allegro](https://developer.allegro.pl/news/zmiany-w-wystawianiu-ofert-dodamy-mozliwosc-wyswietlania-ofert-w-zagranicznych-domenach-serwisu-allegro-VLm1P0KVlhd) wdrożymy także na zasobach **/sale/offers**.

**Dlaczego decydujemy się na wyłączenie zasobów?**

Od początku 2019 roku grupujemy oferty z tym samym produktem, dzięki czemu kupujący łatwiej znajdzie oferty, które spełniają jego oczekiwania. Zwiększamy stopniowo wymagany udział ofert, które są połączone z Katalogiem produktów w wybranych kategoriach, a docelowo wprowadzimy ten wymóg dla wszystkich ofert.

Zasoby **/sale/product-offers** stworzyliśmy z myślą o tym stanie docelowym - wprowadziliśmy w nich istotne usprawnienia przy wystawianiu i edycji ofert powiązanych z produktem. Dlatego zdecydowaliśmy, że będziemy utrzymywać i rozwijać jeden spójny interfejs do zarządzania ofertami z produktem. 

Więcej informacji o zasobach **/sale/product-offers** znajdziesz w [naszym poradniku](https://developer.allegro.pl/tutorials/jak-jednym-requestem-wystawic-oferte-powiazana-z-produktem-D7Kj9gw4xFA).

-----
We would like to remind you that, [as announced](https://developer.allegro.pl/news/at-the-beginning-of-2024-we-will-disable-the-sale-offers-resources-for-creating-and-editing-offers-k1dG88KlxHv), we plan to completely disable resources for creating and editing offers from the **/sale/offers** and **/sale/offers/{offerId}** paths, therefore:
- on **January 1, 2023**, we will mark them as **Deprecated** at documentation,
- on **January 1, 2024** we will **remove** them completely.

In January this year, we have released the **/sale/product-offers** resources in the public version, which allows you to:
- [list an offer assigned to a product with one request](https://developer.allegro.pl/product_offers/),
- [edit any field of the offer](https://developer.allegro.pl/tutorials/how-to-process-list-of-offers-m09BKA5v8H3#editing-single-offer),
- [create a product set](https://developer.allegro.pl/tutorials/list-offer-assigned-product-one-request-D7Kj9M71Bu6#how-to-create-a-product-set).

Only the **/sale/product-offers** resources will allow you to list and edit offers. Ensure that your software uses the equivalent of these resources on the **/sale/product-offers** path before we completely disable the **/sale/offers** resources:

| **Endpoint that we will remove** | **Equivalent to endpoint** | 
| :--------: | :--------: | 
| [POST /sale/offers](https://developer.allegro.pl/documentation#operation/createOfferUsingPOST) | [POST /sale/product-offers](https://developer.allegro.pl/documentation#operation/createProductOffers) | 
| [PUT /sale/offers/{offerId}](https://developer.allegro.pl/documentation#operation/updateOfferUsingPUT) | [PATCH /sale/product-offers/{offerId}](https://developer.allegro.pl/documentation#operation/editProductOffers) | 
| [GET /sale/offers/{offerId}](https://developer.allegro.pl/documentation#operation/getOfferUsingGET) | [GET /sale/product-offers/{offerId}](https://developer.allegro.pl/documentation#operation/getProductOffer) |

**Note!** We will **not disable** the resource for getting the offer list ([GET /sale/offers](https://developer.allegro.pl/documentation#operation/searchOffersUsingGET)) - it will still be available. 

We are constantly working on developing **/sale/product-offers** resources. We verify and complete all missing features in relation to **/sale/offers**, or introduce new resources that will ensure continuity of access to functionality. 

**Will the /sale/offers resources allow offers to be displayed and managed on foreign domains?**

We will also implement the [possibility to display listings in foreign Allegro domains](https://developer.allegro.pl/news/changes-in-listing-offers-we-will-add-the-possibility-to-display-offers-in-foreign-domains-of-allegro-s-marketplaces-9gKv0RVzRTK) on the /sale/offers resources.

**Why do we choose to turn off resources?**

From the beginning of 2019, we group offers with the same product, making it easier for buyers to find offers that meet their needs. We are gradually increasing the required share of offers that are linked to the Product Catalog in selected categories, and ultimately we will introduce this requirement for all offers.

We created the **/sale/product-offers** resources with this target state in mind - we have introduced significant improvements in listing and editing offers assigned to the product. That is why we decided to maintain and develop one consistent interface for managing offers with the product.

For more information about **/sale/product-offers** resources, see [our guide](https://developer.allegro.pl/tutorials/list-offer-assigned-product-one-request-D7Kj9M71Bu6).

**Comment by rafaljosko** [comment]
@AureliuszBrussy   - Czyli mamy czas do 01.2024, by przepisać aplikację ?  Czy od 01.2023 usuwacie ?  Bo już mi każdy suszy, że do końca grudnia muszę przejść :)

**Comment by AureliuszBrussy** [comment]
Tak, masz czas do 1 stycznia 2024 roku - wtedy nastąpi ostateczne wyłączenie zasobów **/sale/offers**.

**Comment by navlukaszzborek** [comment]
Mam podobnie, każdy mówi ze to przestanie działać.
Czyli po nowym roku można normalnie korzystać z /sale/offers do tworzenia ofert oraz /sale/offers/{id} do modyfikowania

**Comment by AureliuszBrussy** [comment]
Tak, jeszcze w 2023 roku zasoby **/sale/offers** będą dostępne, natomiast zalecamy jak najszybsze przejście na zasoby **/sale/product-offers**. Na początku 2023 zasoby **/sale/offers** zostaną oznaczone jako **deprecated**. Oznacza to, że w przyszłym roku potencjalnie możemy wdrożyć nowe funkcjonalności dot. ofert tylko na zasobie **/sale/product-offers**, pozostawiając **/sale/offers** bez zmian.
